### ISSUE TITLE
[🐛 Bug]: allure report does not contain a failed test with `Failed to create session.`

### ISSUE THREAD

**Issue by egerix** [webdriverio/webdriverio#11425]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.18.2

### Node.js Version

v18.14.2

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
{
browserName: 'chrome'
},
```

### What happened?

If a browser session fails to start when running a test, the result of that test will not appear in the report. We run each test in its own worker in Selenoid, and sometimes we may face issues with starting sessions. As a result, the report (for sample: allure) will not contain the specific test.

I know what the behavior is by design. Wdio starts the framework only after session initialization, but I would like to have a correct report even in such cases.

### What is your expected behavior?

The report should contain dropped tests that failed to run due to a session start error

### How to reproduce the bug.

https://github.com/egerix/wdio-bug-sample

`npm run wdio `

The allure-results folder will be empty

### Relevant log output

```typescript
[0-0] 2023-10-16T09:46:26.505Z ERROR webdriver: Request failed with status undefined due to TimeoutError: Timeout awaiting 'response' for 10ms when running "http://0.0.0.0:52000/session" with method "POST" and args "{"capabilities":{"alwaysMatch":{"browserName":"chrome","goog:chromeOptions":{"binary":"/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"}},"firstMatch":[{}]},"desiredCapabilities":{"browserName":"chrome","goog:chromeOptions":{"binary":"/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"}}}"
[0-0] 2023-10-16T09:46:26.505Z ERROR webdriver: TimeoutError: Timeout awaiting 'response' for 10ms when running "http://0.0.0.0:52000/session" with method "POST" and args "{"capabilities":{"alwaysMatch":{"browserName":"chrome","goog:chromeOptions":{"binary":"/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"}},"firstMatch":[{}]},"desiredCapabilities":{"browserName":"chrome","goog:chromeOptions":{"binary":"/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"}}}"
[0-0]     at getTimeoutError (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/webdriver/build/utils.js:317:24)
[0-0]     at NodeJSRequest._request (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/webdriver/build/request/index.js:152:31)
[0-0]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[0-0]     at async startWebDriverSession (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/webdriver/build/utils.js:63:20)
[0-0]     at async WebDriver.newSession (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/webdriver/build/index.js:27:45)
[0-0]     at async remote (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/webdriverio/build/index.js:45:22)
[0-0]     at async Runner._startSession (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/@wdio/runner/build/index.js:224:29)
[0-0]     at async Runner._initSession (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/@wdio/runner/build/index.js:190:25)
[0-0]     at async Runner.run (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/@wdio/runner/build/index.js:79:19)
[0-0] 2023-10-16T09:46:26.506Z ERROR @wdio/runner: Error: Failed to create session.
[0-0] Timeout awaiting 'response' for 10ms when running "http://0.0.0.0:52000/session" with method "POST" and args "{"capabilities":{"alwaysMatch":{"browserName":"chrome","goog:chromeOptions":{"binary":"/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"}},"firstMatch":[{}]},"desiredCapabilities":{"browserName":"chrome","goog:chromeOptions":{"binary":"/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"}}}"
[0-0]     at startWebDriverSession (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/webdriver/build/utils.js:68:15)
[0-0]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[0-0]     at async WebDriver.newSession (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/webdriver/build/index.js:27:45)
[0-0]     at async remote (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/webdriverio/build/index.js:45:22)
[0-0]     at async Runner._startSession (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/@wdio/runner/build/index.js:224:29)
[0-0]     at async Runner._initSession (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/@wdio/runner/build/index.js:190:25)
[0-0]     at async Runner.run (file:///Users/egexiii/Work/wdio-bug-sample/node_modules/@wdio/runner/build/index.js:79:19)
[0-0] FAILED in chrome - file:///features/login.feature
2023-10-16T09:46:26.613Z INFO @wdio/cli:launcher: Run onWorkerEnd hook
2023-10-16T09:46:26.614Z INFO @wdio/cli:launcher: Run onComplete hook

Spec Files:      0 passed, 1 failed, 1 total (100% completed) in 00:00:02 

2023-10-16T09:46:26.615Z INFO @wdio/local-runner: Shutting down spawned worker
2023-10-16T09:46:26.867Z INFO @wdio/local-runner: Waiting for 0 to shut down gracefully
2023-10-16T09:46:26.868Z INFO @wdio/local-runner: shutting down
➜  wdio-bug-sample git:(main) ✗ ls allure-results 
➜  wdio-bug-sample git:(main) ✗
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Thanks for reporting!



We greatly appreciate any contributions that help resolve the bug. While we understand that active contributors have their own priorities, we kindly request your assistance if you rely on this bug being fixed. We encourage you to take a look at our [contribution guidelines](https://webdriver.io/docs/contribute) or join our friendly [Discord](http://discord.webdriver.io/) development server, where you can ask any questions you may have. Thank you for your support, and cheers!

**Comment by Pawel1894** [comment]
Hey, I can give it a shot. Can you assign me?

**Comment by Pawel1894** [comment]
Update - I checked allure report code and I don't see solution for that, atleast I won't be able to do that. 
It failes before any test starts, so it kind of makes impossible to output any info about what happened by design.

**Comment by christian-bromann** [comment]
I looked into the repro and it seems the config has set `connectionRetryTimeout: 10,` which means that a response is expected within 10ms. Even for local execution this is way to short and the reason why the tests fail as the session request times out before the browser driver was able to respond. I recommend set the `connectionRetryTimeout` option to at least `1000`.

Let me know if you have further questions.

**Comment by egerix** [comment]
> I looked into the repro and it seems the config has set `connectionRetryTimeout: 10,` which means that a response is expected within 10ms. Even for local execution this is way to short and the reason why the tests fail as the session request times out before the browser driver was able to respond. I recommend set the `connectionRetryTimeout` option to at least `1000`.
> 
> Let me know if you have further questions.

This timeout is for an example at which the problem repeats. In the case of remote browser startup, we may have situations where sessions are not picked up even for longer timeouts. We don't learn about dropped tests from the report, they just don't get there.

**Comment by christian-bromann** [comment]
Gotcha, I created a new issue #11583 since it is not solely an issue with the Allure report but a very general one. Contributions are welcome!
